nexuse2e-core/src/main/java/org/nexuse2e/ui2/rest/TransactionReportingHandler.java

```diff
@@ -299,4 +328,68 @@ private void returnChoreographyIds(HttpServletResponse response) throws IOExcept
 
         response.getOutputStream().print(new Gson().toJson(choreographyIds));
     }
+
+    private void handleEngineLogRequest(HttpServletRequest request, HttpServletResponse response, boolean count) throws Exception {
+        String pageIndex = request.getParameter("pageIndex");
+        String itemsPerPage = request.getParameter("itemsPerPage");
+        String messageText = request.getParameter("messageText");
+        String severity;
+        Date startDate;
+        Date endDate;
+
+        try {
+            startDate = getDateFromString(request.getParameter("startDate"));
+            endDate = getDateFromString(request.getParameter("endDate"));
+            severity = getSeverityLevelFromName(request.getParameter("severity"));
+        } catch (Exception e) {
+            if (count) {
```

Comment:
I get the feeling that, having a boolean that changes the behavior of this method so much, makes it more cumbersome then it helps. (eg we have the itemsPerPage var that only is needed for the logs) Please test if having to separate methods are cleaner to read.

Reply:
We have decided for the lesser evil because the other option would cause a lot of code duplication. To extract the parameter processing we would need custom data type which is probably too much.